src/som/interpreter/actors/ResolvePromiseNode.java

```diff
@@ -37,23 +37,23 @@ public SResolver selfResolution(final SResolver resolver, final SPromise result,
   /**
    * Handle the case that a promise is resolved with another promise, which is not itself.
    */
-  @Specialization(guards = {"resolver.getPromise() != result"})
-  public SResolver chainedPromise(final VirtualFrame frame,
-      final SResolver resolver, final SPromise result,
+  @Specialization(guards = {"resolver.getPromise() != resolvedValue"})
+  public SResolver chainedPromise(final VirtualFrame frame, final SResolver resolver, final SPromise resolvedValue,
       final boolean isBreakpointOnPromiseResolution) {
     assert resolver.assertNotCompleted();
-    SPromise promise = resolver.getPromise();
+    SPromise promiseToBeResolved = resolver.getPromise();
 
-    synchronized (result) {
-      if (result.isResolvedUnsync()) {
-        normalResolution(frame, resolver, result.getValueUnsync(),
-            isBreakpointOnPromiseResolution);
-      } else if (result.isErroredUnsync()) {
+    synchronized (resolvedValue) {
+      if (resolvedValue.isResolvedUnsync()) { //resolved
```

Comment:
resolved what? also the code is pretty clear, it tests whether the 'value' is `isResolved`.
I still don't like the variable name for `resolvedValue` or `resolutionValue`. Would be good if this reads like a promise name.